lib/fluent/plugin/filter_deparser.rb

```diff
@@ -0,0 +1,35 @@
+class Fluent::DeparserFilter < Fluent::Output
+  Fluent::Plugin.register_filter('deparser', self)
+
+  config_param :format, :string
+  config_param :format_key_names, :string
+  config_param :key_name, :string, :default => 'message'
+  config_param :reserve_data, :bool, :default => false
+
+  def configure(conf)
+    super
+
+    @format_key_names = @format_key_names.split(',')
+    begin
+      dummy = @format % (["x"] * @format_key_names.length)
+    rescue ArgumentError
+      raise Fluent::ConfigError, "mismatch between placeholder of format and format_key_names"
+    end
+  end
+
+  def filter_stream(tag, es)
+    new_es = Fluent::MultiEventStream.new
+    if @reserve_data
+      es.each {|time,record|
+        record.update({@key_name => (@format % @format_key_names.map{|k| record[k]})})
```

Comment:
Do not `update` original record. It breaks records which are read from other plugins.
Use `new_es.add(time, record.merge({...})`.